third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1134,12 +950,26 @@ struct FpToFpOpConversion
     }
     if (useFP16IntermediateSrc)
       for (Value &v : inVals)
-        v = convertFp32ToFp16NZ(loc, rewriter, v);
+        v = cvtFp32ToFp16(loc, rewriter, v,
+                          roundingMode.value_or(RoundingMode::RTNE));
     inVals.resize(numElements, undef(typeConverter->convertType(srcType)));
     SmallVector<Value> outVals;
     if (srcType != dstType) {
-      auto cvtFunc = getConversionFunc(srcType, dstType);
-      outVals = cvtFunc(loc, rewriter, inVals);
+      auto getCvtFunc = getConversionFunc(srcType, dstType, roundingMode);
+      if (failed(getCvtFunc)) {
+        std::string err_msg;
+        llvm::raw_string_ostream errs(err_msg);
+        errs << "Unsupported conversion from " << srcType << " to " << dstType;
+        if (roundingMode.has_value())
+          errs << " with rounding mode "
+               << stringifyRoundingMode(roundingMode.value());
+        errs << "\n";
+        mlir::emitError(loc, err_msg);
+        llvm::report_fatal_error("Unsupported rounding mode for conversion.");
```

Comment:
Is there really a need to call into `llvm::report_fatal_error` to crash the whole compiler? Can we just return an empty vector here to let the pass fail normally?

Reply:
I agree it is better not to crash the whole compiler just for this error. Changed.